### PR TITLE
Update new client input details

### PR DESCRIPTION
**Pull request by palant-dev**
Finished the task of linking text fields to the ViewModel.
Changes:
- added text field with the possibility of using some steppers too.
- added the Array of 6 emojis (can be changed) and an associated picker
- added the sex picker
- added BMI
- added select a goal picker (still with raw values as the sex picker, needs to be improved in the future)

image

### CODE REVIEW

File: SlimApp/NewClient/NewClientView.swift [hunk 1]
```diff
@@ -9,34 +9,69 @@ import SwiftUI
...
+                        TextField("0", value: $viewModel.height, formatter: decimalFormatter)
+                            .keyboardType(.numbersAndPunctuation)
+                        Stepper("Insert Height", value: $viewModel.height, step: 10)
+                            .labelsHidden()
+                    }
+                    HStack {
+                        Text("BMI")
+                        TextField("0", value: $viewModel.bodyMassIndex, formatter: decimalFormatter)
+                            .keyboardType(.numbersAndPunctuation)
+                        Stepper("Insert BMI", value: $viewModel.bodyMassIndex, step: 0.5)
```
Comment (giiks131) [inline SlimApp/NewClient/NewClientView.swift]: wouldn't it be better to calculate BMI?

File: SlimApp/NewClient/NewClientView.swift [hunk 2]
```diff
@@ -9,34 +9,69 @@ import SwiftUI
...
                 Section {
-                    TextField("Height", text: $viewModel.height)
-                        .keyboardType(.decimalPad)
-                    TextField("Weight", text: $viewModel.weight)
-                        .keyboardType(.numberPad)
+                    HStack {
+                        Text("Weight (Kg)")
+                        TextField("0", value: $viewModel.weight, formatter: decimalFormatter)
+                            .keyboardType(.numbersAndPunctuation)
+                        Stepper("Insert Weight", value: $viewModel.weight, step: 0.5)
```
Comment (giiks131) [inline SlimApp/NewClient/NewClientView.swift]: negative values ​​should be limited

File: SlimApp/NewClient/NewClientView.swift [hunk 3]
```diff
@@ -9,34 +9,69 @@ import SwiftUI
...
+                            .labelsHidden()
+                        
+                    }
+                    HStack {
+                        Text("Height (cm)")
+                        TextField("0", value: $viewModel.height, formatter: decimalFormatter)
+                            .keyboardType(.numbersAndPunctuation)
+                        Stepper("Insert Height", value: $viewModel.height, step: 10)
```
Comment (giiks131) [inline SlimApp/NewClient/NewClientView.swift]: negative values ​​should be limited

File: SlimApp/NewClient/NewClientView.swift (hunk 2 above)
Comment (giiks131) [reply SlimApp/NewClient/NewClientView.swift]: also maybe it is better to provide base value which is closer to the average (not 0)

File: SlimApp/NewClient/NewClientView.swift (hunk 3 above)
Comment (giiks131) [reply SlimApp/NewClient/NewClientView.swift]: also maybe it is better to provide base value which is closer to the average

**Review by giiks131 — CHANGES_REQUESTED** [review]
everything seems ok for now, but maybe it is better to fix little problems now instead leaving it for later

File: SlimApp/NewClient/NewClientView.swift (hunk 1 above)
Comment (palant-dev) [reply SlimApp/NewClient/NewClientView.swift]: ok boss!

File: SlimApp/NewClient/NewClientView.swift (hunk 2 above)
Comment (palant-dev) [reply SlimApp/NewClient/NewClientView.swift]: After having a talk with about the stepper and due to some troubles implementing it with bmi and behaviour of the textfields, I decided to remove the stepper so now I think that having 0 value as default for the textfield should be ok

File: SlimApp/NewClient/NewClientView.swift (hunk 3 above)
Comment (palant-dev) [reply SlimApp/NewClient/NewClientView.swift]: After having a talk with about the stepper and due to some troubles implementing it with bmi and behaviour of the textfields, I decided to remove the stepper so now I think that having 0 value as default for the textfield should be ok

File: SlimApp/NewClient/NewClientView.swift (hunk 1 above)
Comment (giiks131) [reply SlimApp/NewClient/NewClientView.swift]: need to fix bug, will discuss it in person. but for now it is ok to merge branches and fix it after this

**Review by giiks131 — APPROVED** [review]
need to fix bug with BMI, will discuss it in person. but for now it is ok to merge branches and fix it after this

**Review by giiks131 — APPROVED** [review]

**Pull request merged**